### PR TITLE
[ADD] l10n_mz: add COA, taxes and tax report

### PR DESCRIPTION
**Pull request by moerradi**
Add chart of accounts, taxes and tax report for Mozambican localization

Community-PR: https://github.com/odoo/odoo/pull/113546
Enterprise-PR: https://github.com/odoo/enterprise/pull/37732

Task-Id: 3162293

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_mz/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+{
+    'name': "Mozambique - Accounting",
+    'description': """
+		Mozambican Accounting localization
+    """,
+    'version': '1.0',
+    'category': 'Accounting/Localizations/Account Charts',
+
```
Comment (rugo-odoo) [inline addons/l10n_mz/__manifest__.py:11]: No need need for blank lines like this in the manifest. Idem for the one a little lower.

File: addons/l10n_mz/data/account.account.template.csv [hunk 2]
```diff
@@ -0,0 +1,319 @@
+"id","name","code","account_type","chart_template_id/id","reconcile"
+"mz_coa_211","Purchased Goods","211","asset_current","l10n_mz.l10nmz_chart_template","False"
```
Comment (rugo-odoo) [inline addons/l10n_mz/data/account.account.template.csv:2]:
```suggestion
"l10n_mz_account_211","Purchased Goods","211","asset_current","l10n_mz.l10nmz_chart_template","False"
```
This is the recent convention for the ids of accounts.

File: addons/l10n_mz/data/account.group.template.csv [hunk 3]
```diff
@@ -0,0 +1,107 @@
+id,code_prefix_start,code_prefix_end,name,chart_template_id/id
+"mz_group_1",1,," Financial resources","l10n_mz.l10nmz_chart_template"
```
Comment (rugo-odoo) [inline addons/l10n_mz/data/account.group.template.csv:2]:
```suggestion
"l10n_mz_account_group_1",1,," Financial resources","l10n_mz.l10nmz_chart_template"
```
This is the recent convention for the ids of account groups.

File: addons/l10n_mz/data/account_chart_template_data.xml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+         <!-- Account Tags -->
```
Comment (rugo-odoo) [inline addons/l10n_mz/data/account_chart_template_data.xml:4]: Don't really need this comment.

File: addons/l10n_mz/data/account_chart_template_data.xml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+        <record id="l10nmz_chart_template" model="account.chart.template">
```
Comment (rugo-odoo) [inline addons/l10n_mz/data/account_chart_template_data.xml:5]:
```suggestion
        <record id="l10n_mz_chart_template" model="account.chart.template">
```

File: addons/l10n_mz/data/account_chart_template_data.xml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+            <field name="name">Plano General de Contabilidad - Pequenas Empresas</field>
```
Comment (rugo-odoo) [inline addons/l10n_mz/data/account_chart_template_data.xml:6]: This should be in english.

File: addons/l10n_mz/data/account_fiscal_position_data.xml [hunk 7]
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="1">
+        <!-- Fiscal Position Templates -->
```
Comment (rugo-odoo) [inline addons/l10n_mz/data/account_fiscal_position_data.xml:4]: Don't need this comment.

File: addons/l10n_mz/data/account_fiscal_position_data.xml [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+        <record id="fiscal_position_import" model="account.fiscal.position.template">
+            <field name="sequence">1</field>
+            <field name="name">Export</field>
+            <field name="chart_template_id" ref="l10nmz_chart_template"/>
+            <field name="auto_apply" eval="True"/>
+        </record>
+
+        <record id="fiscal_position_tax_template_1" model="account.fiscal.position.tax.template">
+            <field name="position_id" ref="fiscal_position_import"  />
```
Comment (rugo-odoo) [inline addons/l10n_mz/data/account_fiscal_position_data.xml:20]:
```suggestion
            <field name="position_id" ref="fiscal_position_import"/>
```

File: addons/l10n_mz/data/account_pgcpe_mozambique.xml [hunk 9]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+        <record id="l10nmz_chart_template" model="account.chart.template">
+            <field name="name">Plano General de Contabilidad - Pequenas Empresas</field>
```
Comment (rugo-odoo) [inline addons/l10n_mz/data/account_pgcpe_mozambique.xml:4]: English.

File: addons/l10n_mz/data/account_tax_group_data.xml [hunk 10]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="1">
+        <record id="tax_group_iva_0" model="account.tax.group">
+            <field name="name">IVA 0%</field>
+            <field name="country_id" ref="base.mz"/>
+        </record>
```
Comment (rugo-odoo) [inline addons/l10n_mz/data/account_tax_group_data.xml:7]:
```suggestion
        <record id="tax_group_vat_0" model="account.tax.group">
            <field name="name">VAT 0%</field>
            <field name="country_id" ref="base.mz"/>
        </record>
```
English.

File: addons/l10n_mz/data/account_fiscal_position_data.xml [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+            <field name="tax_src_id" ref="iva_sale_16" />
```
Comment (rugo-odoo) [inline addons/l10n_mz/data/account_fiscal_position_data.xml:21]:
```suggestion
            <field name="tax_src_id" ref="vat_sale_16" />
```
Use the English abbreviation.

File: addons/l10n_mz/data/account_fiscal_position_data.xml [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+        </record>
+
+        <record id="fiscal_position_import" model="account.fiscal.position.template">
+            <field name="sequence">1</field>
+            <field name="name">Export</field>
+            <field name="chart_template_id" ref="l10nmz_chart_template"/>
+            <field name="auto_apply" eval="True"/>
+        </record>
+
+        <record id="fiscal_position_tax_template_1" model="account.fiscal.position.tax.template">
```
Comment (rugo-odoo) [inline addons/l10n_mz/data/account_fiscal_position_data.xml:19]: Use a more explicit id. What does `template_1` mean ?

File: addons/l10n_mz/data/tax_report.xml [hunk 13]
```diff
@@ -0,0 +1,217 @@
...
+                            </record>
+                        </field>
+                    </record>
+                    <record id="l10n_mz_tax_report_line_sum" model="account.report.line">
+                        <field name="name">Settlement Amount</field>
+                        <field name="children_ids">
+                            <record id="l10n_mz_tax_report_field_11" model="account.report.line">
+                                <field name="name">Field 11 - Tax base Total</field>
+                                <field name="expression_ids">
+                                    <record id="l10n_mz_tax_report_11_tag" model="account.report.expression">
```
Comment (rugo-odoo) [inline addons/l10n_mz/data/tax_report.xml:155]: I think you can use the shorthand version here (for aggregation).

File: addons/l10n_mz/data/tax_report.xml [hunk 14]
```diff
@@ -0,0 +1,217 @@
...
+                            </record>
+                            <record id="l10n_mz_tax_report_field_12" model="account.report.line">
+                                <field name="name">Field 12 - Total tax in favor of the taxable person</field>
+                                <field name="code">MZ_TR_F_12</field>
+                                <field name="expression_ids">
+                                    <record id="l10n_mz_tax_report_12_tag" model="account.report.expression">
+                                        <field name="label">balance</field>
+                                        <field name="engine">aggregation</field>
+                                        <field name="formula">MZ_TR_F_5.balance + MZ_TR_F_6.balance + MZ_TR_F_7.balance + MZ_TR_F_8.balance + MZ_TR_F_9.balance</field>
+                                    </record>
```
Comment (rugo-odoo) [inline addons/l10n_mz/data/tax_report.xml:170]: Same.

File: addons/l10n_mz/data/tax_report.xml [hunk 15]
```diff
@@ -0,0 +1,217 @@
...
+                            </record>
+                            <record id="l10n_mz_tax_report_field_13" model="account.report.line">
+                                <field name="name">Field 13 - Total tax in favor of the state</field>
+                                <field name="code">MZ_TR_F_13</field>
+                                <field name="expression_ids">
+                                    <record id="l10n_mz_tax_report_13_tag" model="account.report.expression">
+                                        <field name="label">balance</field>
+                                        <field name="engine">aggregation</field>
+                                        <field name="formula">MZ_TR_F_2.balance + MZ_TR_F_10.balance</field>
+                                    </record>
```
Comment (rugo-odoo) [inline addons/l10n_mz/data/tax_report.xml:181]: Same same.

File: addons/l10n_mz/data/tax_report.xml [hunk 16]
```diff
@@ -0,0 +1,217 @@
...
+                                </field>
+                            </record>
+                        </field>
+                    </record>
+                    <record id="l10n_mz_tax_report_line_final" model="account.report.line">
```
Comment (rugo-odoo) [inline addons/l10n_mz/data/tax_report.xml:186]: Why not `l10n_mz_tax_report_field_14` ?

File: addons/l10n_mz/data/account_pgcpe_mozambique.xml (hunk 9 above)
Comment (rugo-odoo) [reply addons/l10n_mz/data/account_pgcpe_mozambique.xml:4]: Up.

File: addons/l10n_mz/data/tax_report.xml (hunk 16 above)
Comment (moerradi) [reply addons/l10n_mz/data/tax_report.xml:186]: nah final contains feild 14 and feild 15

**Pull request closed without merging**